giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/core/service/processing/impl/CompletedStorageRequestProcessor.java

```diff
@@ -85,4 +87,14 @@ public String getProcessedTopic() {
     public Class<? extends ICompletedStorageRequest> getRequestClass() {
         return CompletedStorageRequest.class;
     }
+    
+    private void setOldFileVersionsIfReprocessing(IFile file) {
+        if (file.getStorageId() != null && !file.getStorageId().isEmpty()) {
+            if (file.getOldFileVersionIds() == null) {
+                file.setOldFileVersionIds(new ArrayList<>());
+            }
+            file.getOldFileVersionIds().add(file.getStorageId());
+            file.setOldFileVersionIds(file.getOldFileVersionIds());
```

Comment:
still don't understand what this is for. you are setting `oldFileVersionIds` to the value of `oldFileVersionIds` (you're replacing a list with itself. Why?

Reply:
I'm adding the storageId currently stored to the oldFileVersionIds list and updating this list using the set function. 
Post this the storage id will be updated with the new storage id received in the request.

Reply:
but this updates the list:
`file.getOldFileVersionIds().add(file.getStorageId());`
this
`file.setOldFileVersionIds(file.getOldFileVersionIds());`
just sets the list to the same object.